src/server/HelixServer.js

```diff
@@ -148,17 +163,35 @@ export default class HelixServer extends EventEmitter {
     }
     log.info(`Starting franklin-simulator v${packageJson.version}`);
     await new Promise((resolve, reject) => {
-      this._server = this._app.listen(this._port, (err) => {
-        if (err) {
-          reject(new Error(`Error while starting http server: ${err}`));
-        }
-        this._port = this._server.address().port;
-        log.info(`Local Franklin Dev server up and running: http://localhost:${this._port}/`);
-        if (this._project.proxyUrl) {
-          log.info(`Enabled reverse proxy to ${this._project.proxyUrl}`);
-        }
-        resolve();
-      });
+      if (this._tls) {
+        this._server = https.createServer({
+          key: this._key,
+          cert: this._cert,
+        }, this._app);
+        this._server.listen(this._port, (err) => {
+          if (err) {
+            reject(new Error(`Error while starting https server: ${err}`));
+          }
+          this._port = this._server.address().port;
+          log.info(`Local Franklin Dev server up and running: https://localhost:${this._port}/`);
+          if (this._project.proxyUrl) {
+            log.info(`Enabled reverse proxy to ${this._project.proxyUrl}`);
+          }
+          resolve();
+        });
+      } else {
+        this._server = this._app.listen(this._port, (err) => {
+          if (err) {
+            reject(new Error(`Error while starting http server: ${err}`));
+          }
+          this._port = this._server.address().port;
+          log.info(`Local Franklin Dev server up and running: http://localhost:${this._port}/`);
+          if (this._project.proxyUrl) {
+            log.info(`Enabled reverse proxy to ${this._project.proxyUrl}`);
+          }
+          resolve();
+        });
+      }
```

Comment:
that's a lot of duplicate code. I'm sure we can re-use alot

Reply:
Sure, I'm happy to reduce the code.  Let me push a fix.

Reply:
Reformatted code to re-use the listener callback.